### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by marlop92**
Commit initializing project structure with basic TestEntity example. Added H2 and simple controller.

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+## Module description
+
+<p>App working with Java 8</p>
```
Comment (szewa-91) [inline server/README.md]: It's not HTML ;-)

File: server/src/test/java/com/sauron/service/TestEntitySerivceTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    public void invalidIdShouldThrowException() {
+        //given
+        long missingId = 999L;
+
+        //when
+        Mockito.when(testEntityRepo.findTestEntityById(999L))
+                .thenReturn(Optional.empty());
+        Executable action = () -> testEntitySerivce.getTestEntity(missingId);
+
+        //than
```
Comment (szewa-91) [inline server/src/test/java/com/sauron/service/TestEntitySerivceTest.java]: then ;-)

File: server/src/main/java/com/sauron/service/TestEntitySerivce.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.sauron.service;
+
+import com.sauron.model.TestEntityPojo;
+
+public interface TestEntitySerivce {
```
Comment (szewa-91) [inline server/src/main/java/com/sauron/service/TestEntitySerivce.java]: Serivce -> Service

Besides, personally I'm not a big fan of creating interfaces when only one implementation exists (unless there are some reasons behind that)

File: server/src/main/java/com/sauron/model/entities/TestEntity.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "TESTENTITIES")
+public class TestEntity extends BaseEntity{
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", unique = true, updatable = false, nullable = false)
```
Comment (szewa-91) [inline server/src/main/java/com/sauron/model/entities/TestEntity.java]: Do you really need this explicit mapping here? I thing that JPA will know that this column should be unique and not nullable when you use @Id annotation. Specifying name also seems redundant.

File: server/README.md (hunk 1 above)
Comment (szewa-91) [reply server/README.md]: I thing this README needs more love. Describe how to run the project, dependencies, it's not obvious for someone that doesn't have experience in Spring development

File: server/src/test/java/com/sauron/service/TestEntitySerivceTest.java (hunk 2 above)
Comment (marlop92) [reply server/src/test/java/com/sauron/service/TestEntitySerivceTest.java]: Fixed

File: server/src/main/java/com/sauron/service/TestEntitySerivce.java (hunk 3 above)
Comment (marlop92) [reply server/src/main/java/com/sauron/service/TestEntitySerivce.java]: Name changed.

**Pull request merged**